script.js

```diff
@@ -0,0 +1,141 @@
+class ProductAPI {
+  constructor(baseURL) {
+    this.baseURL = baseURL;
+  }
+
+  async apiGet(limit, offset) {
+    try {
+      const response = await fetch(
+        `${this.baseURL}?limit=${limit}&skip=${offset}&select=title,price`
+      );
+      if (!response.ok) {
+        throw new Error(`Error fetching products: ${response.statusText}`);
+      }
+      const data = await response.json();
+      return data;
+    } catch (error) {
+      console.error("Error fetching products:", error);
+      throw error;
+    }
+  }
+
+  async getProduct(id) {
+    try {
+      const response = await fetch(`${this.baseURL}/${id}`);
+      if (!response.ok) {
+        throw new Error(`Error fetching product: ${response.statusText}`);
+      }
+      const data = await response.json();
+      return data;
+    } catch (error) {
+      console.error("Error fetching product:", error);
+      throw error;
+    }
+  }
+}
+
+const baseURL = "https://dummyjson.com/products";
+const productAPI = new ProductAPI(baseURL);
+let limit = 9;
+let data = [];
+let offset = 0;
+let productDetails = {};
+let currentPage = 1;
+
+async function fetchData() {
+  try {
+    const result = await productAPI.apiGet(limit, offset);
+    data = result.products;
+    window.scrollTo(0, 0);
+    renderProducts();
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+async function getProduct(id) {
+  try {
+    const result = await productAPI.getProduct(id);
+    productDetails[id] = result;
+    renderProductDetails(id);
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+function renderProducts() {
+  const productsContainer = document.getElementById("products-container");
+  productsContainer.innerHTML = "";
+  data.forEach((e) => {
+    const productDiv = document.createElement("div");
+    productDiv.className = "product-div";
+    productDiv.innerHTML = `
+    ${e.title}<br />
+    Price: $${e.price}<br />
+    <button onclick="getProduct(${e.id})">Get Details</button>
+    <div id="details-${e.id}"></div>
+  `;
+    productsContainer.appendChild(productDiv);
+  });
+}
+
+function renderProductDetails(id) {

```

Comment:
method will be the part of the your class

Reply:
done